### PR TITLE
[vectorized](udaf) fix java udaf meet error of std::bad_alloc

### PR DESCRIPTION
**Pull request by zhangstar333**
# Proposed changes
Now if the user code of java udaf throws exception, because c++ code of agg function nobody could deal
with it, so maybe get error of std::bad_alloc

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/java-udf/src/main/java/org/apache/doris/udf/UdafExecutor.java [hunk 1]
```diff
@@ -85,6 +85,7 @@ public void add(boolean isSinglePlace, long rowStart, long rowEnd) throws UdfRun
                 } while (isSinglePlace && idx < rowEnd);
             } while (idx < rowEnd);
         } catch (Exception e) {
+            LOG.info("invoke add function meet some error: " + e.getCause().toString());
```
Comment (Gabriel39) [inline fe/java-udf/src/main/java/org/apache/doris/udf/UdafExecutor.java:88]: LOG.warning

File: be/src/runtime/user_function_cache.cpp [hunk 2]
```diff
@@ -425,43 +425,4 @@ Status UserFunctionCache::get_jarpath(int64_t fid, const std::string& url,
     return Status::OK();
 }
 
-Status UserFunctionCache::check_jar(int64_t fid, const std::string& url,
```
Comment (HappenLee) [inline be/src/runtime/user_function_cache.cpp:428]: why del this code of `check_jar`?

File: docs/en/docs/ecosystem/udf/java-user-defined-function.md [hunk 3]
```diff
@@ -107,13 +107,11 @@ The following SimpleDemo will implement a simple function similar to sum, the in
 ```JAVA
 package org.apache.doris.udf.demo;
 
-import org.apache.hadoop.hive.ql.exec.UDAF;
-
 import java.io.DataInputStream;
 import java.io.DataOutputStream;
 import java.io.IOException;
 
-public class SimpleDemo extends UDAF {
```
Comment (HappenLee) [inline docs/en/docs/ecosystem/udf/java-user-defined-function.md:116]: why not extend the UDAF?

File: be/src/runtime/user_function_cache.cpp (hunk 2 above)
Comment (zhangstar333) [reply be/src/runtime/user_function_cache.cpp:428]: now could use get_jarpath function to check this

File: docs/en/docs/ecosystem/udf/java-user-defined-function.md (hunk 3 above)
Comment (zhangstar333) [reply docs/en/docs/ecosystem/udf/java-user-defined-function.md:116]: the is no need't to extend anything, it's meaningless

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 36.13 seconds
 stream load tsv:          451 seconds loaded 74807831229 Bytes, about 158 MB/s
 stream load json:         25 seconds loaded 2358488459 Bytes, about 89 MB/s
 stream load orc:          73 seconds loaded 1101869774 Bytes, about 14 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230317151257_clickbench_pr_116161.html

**Review by HappenLee — APPROVED** [review]
LGTM

**Comment by yiguolei** [comment]
auto bucket FE UT failed, I skip it since this test case is not stable.

**Pull request merged**